### PR TITLE
chore(ctb): Move Game Duration to Deploy Config

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Change the Fault Dispute Game constant `GAME_DURATION` variable to be immutably set through the constructor so we can read it in from the deploy config on deployment.

Rational: this makes game configuration easier for e2e tests and local challenger testing.

**Metadata**

Fixes CLI-4245

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6351** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6351" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6351?utm_source=stack-comment).

File: packages/contracts-bedrock/contracts/test/FaultDisputeGame.t.sol [hunk 1]
```diff
@@ -29,7 +29,7 @@ contract FaultDisputeGame_Init is DisputeGameFactory_Init {
     function init(Claim rootClaim, Claim absolutePrestate) public {
         super.setUp();
         // Deploy an implementation of the fault game
-        gameImpl = new FaultDisputeGame(absolutePrestate, 4, new AlphabetVM(absolutePrestate));
+        gameImpl = new FaultDisputeGame(absolutePrestate, 4, Duration.wrap(604800), new AlphabetVM(absolutePrestate));
```
Comment (clabby) [inline packages/contracts-bedrock/contracts/test/FaultDisputeGame.t.sol]:
```suggestion
        gameImpl = new FaultDisputeGame(absolutePrestate, 4, Duration.wrap(7 days), new AlphabetVM(absolutePrestate));
```

File: packages/contracts-bedrock/deploy-config/devnetL1.json [hunk 2]
```diff
@@ -45,5 +45,6 @@
   "l1StartingBlockTag": "earliest",
   "l2GenesisRegolithTimeOffset": "0x0",
   "faultGameAbsolutePrestate": 140,
-  "faultGameMaxDepth": 4
-}
\ No newline at end of file
+  "faultGameMaxDepth": 4,
+  "faultGameDuration": 604800
```
Comment (clabby) [inline packages/contracts-bedrock/deploy-config/devnetL1.json]: Maybe name this `faultGameMaxDuration`? A bit more descriptive of what the value actually means.

**Review by clabby — APPROVED** [review]
LGTM; Some small nits, feel free to ignore.

**Review by clabby — APPROVED** [review]
LGTM, :shipit:

**Pull request merged**